File: VB/NavBarAndPrism.ModuleA/ModuleA.vb
Imports System
Imports System.Net
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Documents
Imports System.Windows.Ink
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Animation
Imports System.Windows.Shapes
Imports NavBarAndPrism.Views
Imports System.ComponentModel.Composition
Imports Prism.Mef.Modularity
Imports Prism.Regions
Imports Prism.Modularity

Namespace NavBarAndPrism.Modules
	<ModuleExport(GetType(ModuleA))>
	Public Class ModuleA
		Implements IModule

		<Import>
		Public RegionManager As IRegionManager
        Public Sub Initialize() Implements IModule.Initialize
            RegionManager.RegisterViewWithRegion("NavBarControlRegion", GetType(ModuleAView))
        End Sub
    End Class
End Namespace

File: CS/NavBarAndPrism.ModuleA/ModuleA.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using NavBarAndPrism.Views;
using System.ComponentModel.Composition;
using Prism.Mef.Modularity;
using Prism.Regions;
using Prism.Modularity;

namespace NavBarAndPrism.Modules {
    [ModuleExport(typeof(ModuleA))]
    public class ModuleA : IModule {
        [Import]
        public IRegionManager RegionManager;
        public void Initialize() {
            RegionManager.RegisterViewWithRegion("NavBarControlRegion", typeof(ModuleAView));
        }
    }
}
